src/content/developers/docs/oracles/index.md

```diff
@@ -1,314 +1,86 @@
-     */
-    constructor() public {
-        priceFeed = AggregatorV3Interface(0x9326BFA02ADD2366b30bacB125260Af641031331);
-    }
-
-    /**
-     * Returns the latest price
-     */
-    function getLatestPrice() public view returns (int) {
-        (
-            uint80 roundID,
-            int price,
-            uint startedAt,
-            uint timeStamp,
-            uint80 answeredInRound
-        ) = priceFeed.latestRoundData();
-        return price;
-    }
-}
-```
+## Prerequisites {#prerequisites}
 
-[You can test this in remix with this link](https://remix.ethereum.org/#version=soljson-v0.6.7+commit.b8d736ae.js&optimize=false&evmVersion=null&gist=0c5928a00094810d2ba01fd8d1083581)
+This page assumes the reader is familiar with Ethereum fundamentals, including [nodes](/developers/docs/nodes-and-clients/), [consensus mechanisms](/developers/docs/consensus-mechanisms/), and the [EVM](/developers/docs/evm/). You should also have a good grasp of [smart contracts](/developers/docs/smart-contracts/) and [smart contract anatomy](/developers/docs/smart-contracts/anatomy/), especially events. 
 
-[View the docs](https://docs.chain.link/docs/get-the-latest-price)
+## What is a blockchain oracle? {#what-is-a-blockchain-oracle} 
 
-### Chainlink VRF {#chainlink-vrf}
+Oracles are applications that source, verify, and transmit external information (i.e., information stored off-chain) to smart contracts running on the blockchain. Besides “pulling” off-chain data and broadcasting it on Ethereum, oracles can also “push” information from the blockchain to external systems. An example is an oracle that unlocks a smart lock once the user sends the fee via an Ethereum transaction. 
 
-Chainlink VRF (Verifiable Random Function) is a provably-fair and verifiable source of randomness designed for smart contracts. Smart contract developers can use Chainlink VRF as a tamper-proof random number generation (RNG) to build reliable smart contracts for any applications which rely on unpredictable outcomes:
+Oracles act as a “bridge” connecting smart contracts on blockchains to the real world. Without oracles, smart contract applications would only be able to access on-chain data. A blockchain oracle provides a mechanism for triggering smart contract functions using real-world data.
 
-- Blockchain games and NFTs
-- Random assignment of duties and resources (e.g. randomly assigning judges to cases)
-- Choosing a representative sample for consensus mechanisms
+Oracles differ based on the source of data (one or multiple sources), trust models (centralized or decentralized) and system architecture (immediate-read, publish-subscribe, and request-response). We can also distinguish between oracles based on whether they retrieve external data for use by on-chain contracts (input oracles), send information from the blockchain to the real world (output orales), or perform computational tasks off-chain (computational oracles).
```

Comment:
It feels kinda odd that we are setting up for talking about input, output, and computational oracles here and then don't explicitly comment on these later.

Reply:
We already talk about computational oracles in some sections, especially ones about generating randomness off-chain, e.g., the following line in the oracle node section: 

> Computational oracles also rely on off-chain nodes to perform intensive computational tasks, which would be impractical to execute on-chain, given gas costs and block size limits. For example, the oracle node may be tasked with generating a verifiably random figure (e.g., for blockchain-based games).  

And input oracles are pretty much everything we cover in most parts (especially w/ price oracles and all). I didn't think it was necessary to go into detail concerning each oracle type since the content implicitly suggests the type of oracle in question at different points (e.g., price oracles are input oracles). Moreover, output oracle use-cases are still few and far in between, so I didn't want to dedicate a lot of time to that part.